### PR TITLE
damageprofiler

### LINKED ISSUE #285: new module: damageprofiler
https://github.com/Integrative-Transcriptomics/DamageProfiler

Typical usage:
```
damageprofiler -i bam_file -r reference_fasta -o output_directory
```

Already used in nf-core pipelines such as [Eager](https://nf-co.re/eager), and [coproID](https://nf-co.re/coproid)

### PR DESCRIPTION
**Pull request by jfy133**
## PR checklist

This adds the module for DamageProfiler

Closes #285 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `<SOFTWARE>.version.txt` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

**Comment by jfy133** [comment]
Just waiting for singularity image to be created

File: tests/software/damageprofiler/main.nf [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+    print(input)
+
+    fasta = file(params.test_data['homo_sapiens']['genome']['genome_fasta'], checkIfExists: true)
+
+    print(fasta)
+
+    fai = file(params.test_data['homo_sapiens']['genome']['genome_fasta_fai'], checkIfExists: true)
+
+    print(fai)
```
Comment (KevinMenden) [inline tests/software/damageprofiler/main.nf:20]: I guess those were for debugging? So we could remove them.
(Guilty of forgetting those myself all the time)

File: tests/software/damageprofiler/test.yml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    - path: output/damageprofiler/test/5pCtoT_freq.txt
+      md5sum: fcc48ee5f72edff930d627c8bfdd8a5b
+    - path: output/damageprofiler/test/5p_freq_misincorporations.txt
+      md5sum: 54665474f5ef17dcc268567e5eaa7d86
+    - path: output/damageprofiler/test/DamagePlot_five_prime.svg
+      should_exist: true
+    - path: output/damageprofiler/test/DamagePlot.pdf
+      should_exist: true
+    - path: output/damageprofiler/test/DamagePlot_three_prime.svg
+      should_exist: true
```
Comment (KevinMenden) [inline tests/software/damageprofiler/test.yml:19]: You can omit those `should_exist` statements, once you specified a path, that's basically taken care of. Doesn't hurt to leave them in, just makes the code a bit smaller.

**Review by KevinMenden — APPROVED** [review]
Nice, looks perfectly clean to me! Just two minor things that don't actually break the code :+1:

**Comment by jfy133** [comment]
@KevinMenden I reduced the tags to be biased (:guardswoman: ), if that's Ok shall I merge?

**Comment by KevinMenden** [comment]
Sure this looks like merge-material!

**Comment by apeltzer** [comment]
🤠

**Review by apeltzer — APPROVED** [review]
Looking 🔝 to me

**Pull request merged**